[release/2.14] [split from #4773] Unexpected success in functorch/test_ops_xpu.py: stale xfail entries for max_unpool2d / max_pool2d

### 🐛 Describe the bug

parent_issue: 4773
test_cases:
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float64
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmap_autograd_grad_nn_functional_max_unpool2d_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmap_autograd_grad_nn_functional_max_unpool2d_xpu_float64
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapvjp_nn_functional_max_unpool2d_grad_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapvjp_nn_functional_max_unpool2d_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapvjpvjp_nn_functional_max_unpool2d_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapvjpvjp_nn_functional_max_unpool2d_grad_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapjvpvjp_nn_functional_max_unpool2d_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapjvpvjp_nn_functional_max_unpool2d_grad_xpu_float32
  - op_ut,third_party.torch-xpu-ops.test.xpu.functorch.test_ops_xpu.TestOperatorsXPU,test_vmapvjpvjp_nn_functional_max_pool2d_xpu_float32
---

## Parent Issue

Split from #4773 — [Bug Skip]: [Regression]UT failures 2026-8-2

## Affected Test Cases

- `test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float32`
- `test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float64`
- `test_vmap_autograd_grad_nn_functional_max_unpool2d_xpu_float32`
- `test_vmap_autograd_grad_nn_functional_max_unpool2d_xpu_float64`
- `test_vmapvjp_nn_functional_max_unpool2d_grad_xpu_float32`
- `test_vmapvjp_nn_functional_max_unpool2d_xpu_float32`
- `test_vmapvjpvjp_nn_functional_max_unpool2d_xpu_float32`
- `test_vmapvjpvjp_nn_functional_max_unpool2d_grad_xpu_float32`
- `test_vmapjvpvjp_nn_functional_max_unpool2d_xpu_float32`
- `test_vmapjvpvjp_nn_functional_max_unpool2d_grad_xpu_float32`
- `test_vmapvjpvjp_nn_functional_max_pool2d_xpu_float32`

All 11 in `test/xpu/functorch/test_ops_xpu.py::TestOperatorsXPU`.

## Root Cause

All 11 cases fail with `Unexpected success` — i.e. the tests are marked `xfail` in the XPU copy of the test file but now pass.

`test/xpu/functorch/test_ops_xpu.py` is a fork of upstream `pytorch/test/functorch/test_ops.py`. Upstream removed several `xfail("nn.functional.max_unpool2d")`, `xfail("nn.functional.max_unpool2d", "grad")` and `xfail("nn.functional.max_pool2d")` entries from the `vmap_autograd_grad`, `vmapvjp`, `vmapvjpvjp` and `vmapjvpvjp` decorator lists, but the XPU fork still carries them.

Concretely, comparing the XPU fork with upstream `test/functorch/test_ops.py`:

| XPU fork line | Entry | Present upstream? |
|---|---|---|
| 1038 | `xfail("nn.functional.max_pool2d")` | no |
| 1042 | `xfail("nn.functional.max_unpool2d")` | no |
| 1046 | `xfail("nn.functional.max_unpool2d", "grad")` | no |
| 1211-1212 | `xfail("nn.functional.max_unpool2d")` / `"grad"` | no |
| 2021-2022 | `xfail("nn.functional.max_unpool2d")` / `"grad"` | no |
| 2366-2367 | `xfail("nn.functional.max_unpool2d")` / `"grad"` | no |

The functionality behind these xfails (vmap over `max_unpool2d` / `max_pool2d` transforms) was fixed upstream, so the tests now succeed and `unittest` reports `Unexpected success`. This is a test-side staleness
problem, not an XPU kernel bug.

**Fix**: re-sync the affected decorator lists in
`test/xpu/functorch/test_ops_xpu.py` with upstream `test/functorch/test_ops.py`, removing the obsolete `max_unpool2d` / `max_pool2d` xfail entries.

## Reproducer

For each affected test case, run individually:

```bash
source ~/.bashrc && cd third_party/torch-xpu-ops/test/xpu && python -m pytest -sxv functorch/test_ops_xpu.py -k "test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float32"
```

(substitute the test name for each of the 11 cases)

## Representative Error

```
=================================== FAILURES ===================================
_ TestOperatorsXPU.test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float32 _
Unexpected success
=========================== short test summary info ============================
FAILED functorch/test_ops_xpu.py::TestOperatorsXPU::test_vmap_autograd_grad_nn_functional_max_unpool2d_grad_xpu_float32
===================== 1 failed, 10304 deselected in 11.24s =====================
```

### Versions

- Latest good PyTorch: [f2fa815](https://github.com/pytorch/pytorch/commit/f2fa81593a74bb8aca2c3da562765378665af089)
- Current (failing) PyTorch: [4ee8fea](https://github.com/pytorch/pytorch/commit/4ee8fea1f8cbb27acf0dac1c4f2e8b8bb6aef7f7)
- Platform: Intel(R) Data Center GPU Max 1100, Ubuntu 24.04, libze1 1.24.0.0, intel-opencl-icd 25.18.33578.51
- XPU build: 20260100
- Reproduced locally on `torch 2.14.0a0+gitfdd065a`, torch-xpu-ops `1e6e7fbe`.

--- a/test/xpu/functorch/test_ops_xpu.py
+++ b/test/xpu/functorch/test_ops_xpu.py
@@ -1032,17 +1032,6 @@ class TestOperators(TestCase):
                 xfail(
                     "nn.functional.layer_norm"
                 ),  # vmap: inplace into a regular tensor
-                # RuntimeError: NYI: querying is_contiguous inside of vmap
-                # for memory_format other than torch.contiguous_formats
-                xfail("nn.functional.max_pool2d"),
-                # RuntimeError: NYI: Tensor.clone(memory_format) inside vmap is only
-                # supported with memory_format torch.preserve_format or
-                # torch.contiguous_format (got ChannelsLast)
-                xfail("nn.functional.max_unpool2d"),
-                # RuntimeError: NYI: Tensor.clone(memory_format) inside vmap is only
-                # supported with memory_format torch.preserve_format
-                # or torch.contiguous_format (got ChannelsLast)s
-                xfail("nn.functional.max_unpool2d", "grad"),
                 xfail(
                     "nn.functional.rrelu"
                 ),  # RuntimeError: vmap: we do not yet support aten::rrelu_with_noise.
@@ -1206,9 +1195,6 @@ class TestOperators(TestCase):
             xfail("scatter_reduce", "prod"),  # item call
             # Batching rule not implemented for aten::_use_cudnn_ctc_loss.Tensor
             xfail("nn.functional.ctc_loss", device_type="cuda"),
-            # NYI: querying is_contiguous inside of vmap for memory_format other than torch.contiguous_format
-            xfail("nn.functional.max_unpool2d"),
-            xfail("nn.functional.max_unpool2d", "grad"),
             xfail("sparse.sampled_addmm", ""),
             xfail("sparse.mm", "reduce"),
             xfail("as_strided_scatter", ""),  # calls as_strided
@@ -2011,10 +1997,6 @@ class TestOperators(TestCase):
                 # running_mean or running_var, which will be updated in place,
                 # were not batched.
                 xfail("nn.functional.instance_norm"),
-                # NYI: Tensor.clone(memory_format) inside vmap is only supported with
-                # memory_format torch.preserve_format or torch.contiguous_format (got ChannelsLast)
-                xfail("nn.functional.max_unpool2d"),
-                xfail("nn.functional.max_unpool2d", "grad"),
                 xfail(
                     "nn.functional.multi_margin_loss"
                 ),  # Forward AD not implemented and no decomposition
@@ -2357,8 +2339,6 @@ class TestOperators(TestCase):
         {
             # The size of tensor a (4) must match the size of tensor b (10) at non-singleton dimension 0
             xfail("masked_select"),
-            xfail("nn.functional.max_unpool2d", "grad"),  # contiguous call
-            xfail("nn.functional.max_unpool2d"),  # contiguous call
             xfail("to_sparse"),  # dispatch key issue
             xfail("torch.ops.aten._efficient_attention_forward"),  # outputs ints
             # https://github.com/pytorch/pytorch/issues/96560#issuecomment-2151063723

